applicationset/services/scm_provider/utils.go

```diff
@@ -126,6 +126,9 @@ func getBranches(ctx context.Context, provider SCMProviderService, repos []*Repo
 	for _, repo := range repos {
 		reposFilled, err := provider.GetBranches(ctx, repo)
 		if err != nil {
+			if checkBranchDoNotExist(err.Error()) {
```

Comment:
Would this hide an error where the repo doesn't exist? Or will it only return true if a _branch_ doesn't exist?

Reply:
It will only skip errors if branch doesn't exist, i'll add a UT for this

Reply:
Added Test cases